eePlumB/Z_PrepAOI/TileAOI.Rmd

```diff
@@ -0,0 +1,101 @@
+---
+title: "Tile AOI"
+author: "B Steele"
+date: "2023-03-07"
+output: html_document
+---
+
+# Purpose
+
+This script takes an area of interest (AOI) and creates tiles within the extent for use in eePlumB. While this script is built around Lake Superior, it could easily be adapted to any other lake.
+
+```{r}
+library(tidyverse)
+library(sf)
+
+#point to directory where your shapefile lives
+shp_dir = '/Users/steeleb/Documents/GitHub/Superior-Plume-Bloom/data/aoi/'
+```
+
+# Tile AOI
+
+## Read in AOI shapefile
+
+```{r}
+shape = read_sf(file.path(shp_dir, 'superior_no_harbor.shp'))
+
+#in our case, we need to filter for water
+shape = shape %>% 
+  filter(Feature == 'Water')
+
+#plot quickly to confirm
+plot(shape['Feature'])
+```
+
+## Tile shapefile using `st_make_grid`
+
+```{r}
+tiles = st_make_grid(shape,
+                     cellsize = c(35000, 25000))#units are square meters
+tiles = tiles[shape]
+
+#plot quickly to confirm
+plot(tiles)
+plot(shape['Feature'], add = T)
+```
+
+For our purposes, we're going to remove the top row of tiles.
+
+```{r}
+tiles_sub = tiles[1:12]
+plot(tiles_sub)
+plot(shape['Feature'], add = T)
+```
+
+And we'll clip the AOI to the extent of these boxes.
+
+```{r}
+shape_clip = st_crop(shape, st_bbox(tiles_sub))
+
+plot(tiles_sub)
+plot(shape_clip['Feature'], add = T)
+```
+
+And now, we need to merge tiles 9 and 10 (there's just a sliver of area that we want to have a bit more context to label.
+
+```{r}
+tiles_toMerge = tiles[c(9,10)]
```

Comment:
Should you also merge the two tiles right below 9 and 10 who have a similar sliver issue?

![image](https://user-images.githubusercontent.com/13220910/223746831-e0e26773-4858-484a-ae40-96b32a505ec0.png)

Reply:
It is worth noting that I don't have your updated AOI shapefile without the harbor, so this might be an artifact of the older shp file.

Reply:
Yes! This is 100% an artifact of a different shapefile - this is probably a good excuse to use CUAHSI for reproducibility and a need to document that this won't work for all shapefiles and folks might need to adjust for a more sensical tile of the AOI.

Reply:
This is how it should look - where that very teeny corner of the top row is included with the next tile to the right:
![image](https://user-images.githubusercontent.com/32140074/223823815-df1c533d-943e-48bb-9919-7bddb1000a63.png)dc969bf466e5.png)

I'll work on the reproducibility side at a later impasse!